src/o365/graph/commands/schemaextension/schemaextension-add.ts

```diff
@@ -0,0 +1,244 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import * as request from 'request-promise-native';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  description: string;
+  owner: string;
+  targetTypes: string;
+  properties: string;
+}
+
+class GraphSchemaExtensionAdd extends GraphCommand {
+  public get name(): string {
+    return `${commands.SCHEMAEXTENSION_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Creates a Microsoft Graph schema extension';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = args.options.id;
+    telemetryProps.description = args.options.description;
+    telemetryProps.owner = args.options.owner;
+    telemetryProps.targetTypes = args.options.targetTypes;
+    telemetryProps.properties = args.options.properties;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        if (this.verbose) {
+          cmd.log(`Adding schema extension with id '${args.options.id}'...`);
+        }
+
+        const targetTypes = args.options.targetTypes.split(',').map(t => t.trim());
+        let properties: any = null;
```

Comment:
Just because initially there was a try/catch around the parse method, but since it is already checked in the validate(), I ended up removing here from here :)